R/tt_toString.R

```diff
@@ -590,29 +590,11 @@ setMethod("get_formatted_cells", "ElementaryTable",
 #' @rdname gfc
 setMethod("get_formatted_cells", "TableRow",
           function(obj, shell = FALSE) {
-            default_format <- if (is.null(obj_format(obj))) "xx" else obj_format(obj)
-            format <- lapply(row_cells(obj), function(x) {
-                format <- obj_format(x)
-                if (is.null(format))
-                    default_format
-                else
-                    format
-            })
-            default_na_str <- obj_na_str(obj) %||% "NA"
-            na_str <- vapply(row_cells(obj), function(x) {
-                nastr <- obj_na_str(x)
-                if(is.null(nastr))
-                    default_na_str
-                else
-                    nastr
-            }, "")
-
-            matrix(unlist(Map(function(val, format, spn, na_str) {
+            matrix(unlist(Map(function(val, spn) {
                 stopifnot(is(spn, "integer"))
-                val <- if(shell) format else paste(format_rcell(val, format, na_str = na_str),
-                                                   collapse = ", ")
+                val <- paste(format_rcell(val, shell = shell), collapse = ", ")
                 rep(list(val), spn)
-            }, val = row_values(obj), format = format, spn = row_cspans(obj), na_str = na_str)),
+            }, val = row_cells(obj), spn = row_cspans(obj))),
```

Comment:
The key here (and the reason the tests didn't pass, I suspect), is that this doesn't preserve the inheritance from the row object if the cell object doesnt have a format or an na_str value. 

We need a way to make sure that still happens, even though we're operating on the cells now.

Perhaps accept a `parent_row` argument, which defaults to NULL, but which when not null, provides format and na_str information if and only if the corresponding value on the cell is not set.

Reply:
actually, the error was coming from the fact I was not considering the option by which format is a function. I think I found a way to get rid of `shell` in `format_rcells`